src/huggingface_hub/hub_mixin.py

```diff
@@ -19,30 +19,154 @@
 logger = logging.getLogger(__name__)
 
 
-class ModelHubMixin:
-    def __init__(self, *args, **kwargs):
-        """
-        Mix this class with your torch-model class for ease process of saving & loading from huggingface-hub
+class FileManager:
```

Comment:
Since this object is only used by the Hub Mixin right now, I left this object in this file. Is `hub_mixin.py` the right home for `FileManager`?